components/github-api-client/src/client.rs

```diff
@@ -248,14 +252,32 @@ impl OAuthClient for GitHubClient {
             return Err(OAuthError::HttpResponse(rep.status, body));
         }
 
-        let user: User = serde_json::from_str(&body).map_err(
+        let user: serde_json::Value = serde_json::from_str(&body).map_err(
             OAuthError::Serialization,
         )?;
 
+        // Gitlab API has username instead of login
+        let username = if user["login"].is_string() {
+            user["login"].to_string()
+        } else {
+            assert!(user["username"].is_string());
```

Comment:
What are the circumstances in which this `assert!` can fail? If I'm reading the code right, it's when the Github or Gitlab response contains neither `"login"` nor `"username"`. Is it possible they could change the format of their response? What do we gain by panicking over an error with a message?

Reply:
I should probably add a comment here for a TBD to more cleanly split up the API handling between Github and Gitlab as we do the next round of auth refactor, and maybe go back to the structured deserialization.  I'll do that separately.

Reply:
Yeah I think structured deserialization is the right long term plan.